src/components/DepositWidget/Row.tsx

```diff
@@ -28,19 +31,39 @@ const TokenTr = styled.tr`
 
   &.selected {
     background-color: #ecdcff;
+
+  div.success {
+    color: #63ab52;
+    margin-top: -0.5rem;
   }
 `
 
 export interface RowProps {
-  tokenBalances: TokenBalanceDetails
+  initialTokenBalances: TokenBalanceDetails
 }
 
 function _loadFallbackTokenImage(event: React.FormEvent<HTMLImageElement>): void {
   const image = event.target as HTMLImageElement
   image.src = unknownTokenImg
 }
 
-export const Row: React.FC<RowProps> = ({ tokenBalances }: RowProps) => {
+const txOptionalParams: TxOptionalParams = {
+  onSentTransaction: (receipt: Receipt): void => {
+    toast.info(
+      <div>
+        The transaction has been sent! Check{' '}
+        <a href={`https://etherscan.io/tx/${receipt.transactionHash}`} target="_blank" rel="noopener noreferrer">
+          {abbreviateString(receipt.transactionHash, 6, 4)}
+        </a>{' '}
+        for details
+      </div>,
+    )
+  },
+}
+
+export const Row: React.FC<RowProps> = (props: RowProps) => {
+  const { balances, updateBalances } = useTokenBalances(props.initialTokenBalances.address)
```

Comment:
What I did was to implement approach `2`, although might not have been clear.
I start with the copy of `tokenBalances` from the parent.

> why you listen for props.initialTokenBalances.address changes. Do you expect that to change?

Using `props.tokenBalances.address`, the hook is initialized and only fetch info related to the given token from the `depositApi`.
My main goal here was to avoid querying balances for all tokens when only 1 had any update.
So, whenever there's an action that modifies the balances, I call `updateBalances` to force the state to be updated and trigger a redraw.

But let's say I go for approach `1`. 
On `Row`, the deposit action is performed, then `onDeposit` from the parent is invoked.
What needs to happen on the parent in order to trigger the balances re-fetch and redraw?
Isn't it needed to update the `balances` state somehow?

On my mind, even if I don't make `useTokenBalances` update a single token at a time, I would need the `updateBalances` function.

Reply:
Never mind all that, next comment helped me get it. The new version should address it :)